GVRf/Framework/framework/src/main/java/org/gearvrf/GVRPicker.java

```diff
@@ -277,36 +466,48 @@ public static final float pickSceneObject(GVRSceneObject sceneObject,
      * @param dz
      *            The z vector of the ray direction.
      * @return A list of {@link GVRPickedObject}, sorted by distance from the
-     *         camera rig. Each {@link GVRPickedObject} contains the object
-     *         within the {@link GVREyePointeeHolder} along with the hit
-     *         location. (Note that the hit location is actually the point where
-     *         the cast ray intersected the holder's axis-aligned bounding box,
-     *         which may not be exactly where the ray would intersect the scene
-     *         object itself.)
+     *         camera rig. Each {@link GVRPickedObject} contains the scene object
+     *         which owns the {@link GVRCollider} along with the hit
+     *         location and distance from the camera. 
      * 
      * @since 1.6.6
      */
-    public static final List<GVRPickedObject> findObjects(GVRScene scene, float ox, float oy, float oz, float dx,
+    public static final GVRPickedObject[] pickObjects(GVRScene scene, float ox, float oy, float oz, float dx,
             float dy, float dz) {
-        sFindObjectsLock.lock();
-        try {
-            final GVRContext gvrContext = scene.getGVRContext();
-
-            final long[] pointers = NativePicker.pickScene(scene.getNative(), ox, oy, oz, dx, dy, dz);
-            final List<GVRPickedObject> result = new ArrayList<GVRPickedObject>(pointers.length);
-            for (final long pointer : pointers) {
-                final GVREyePointeeHolder holder = GVREyePointeeHolder.lookup(gvrContext, pointer);
-                if (null != holder) {
-                    result.add(new GVRPickedObject(holder.getOwnerObject(), holder.getHit()));
-                }
-            }
+        sFindObjectsLock.lock();        
+        try {            
+            final GVRPickedObject[] result = NativePicker.pickObjects(scene.getNative(), ox, oy, oz, dx, dy, dz);
             return result;
         } finally {
             sFindObjectsLock.unlock();
         }
     }
 
     /**
+     * Casts a ray into the scene graph, and returns the objects it intersects.
+     * 
+     * @deprecated use GVRPickedObject[] pickObjects
+     */
+    public static final List<GVRPickedObject> findObjects(GVRScene scene, float ox, float oy, float oz, float dx,
+            float dy, float dz) {
+        return Arrays.asList(pickObjects(scene, ox, oy, oz, dx, dy, dz));
+    }
+    
+    /**
+     * Internal utility to help JNI add hit objects to the pick list.
+     */
+    public static GVRPickedObject makeHit(long colliderPointer, float distance, float hitx, float hity, float hitz)
```

Comment:
It is no longer public

From: rahul [mailto:notifications@github.com]
Sent: Thursday, June 23, 2016 2:15 PM
To: Samsung/GearVRf <GearVRf@noreply.github.com>
Cc: Nola Donato <nola.donato@samsung.com>; Author <author@noreply.github.com>
Subject: Re: [Samsung/GearVRf] Picking Reform (#621)


In GVRf/Framework/framework/src/main/java/org/gearvrf/GVRPicker.java<https://github.com/Samsung/GearVRf/pull/621#discussion_r68316348>:

>              return result;

>          } finally {

>              sFindObjectsLock.unlock();

>          }

>      }

>

>      /**

> +     * Casts a ray into the scene graph, and returns the objects it intersects.

> +     *

> +     * @deprecated use GVRPickedObject[] pickObjects

> +     */

> +    public static final List<GVRPickedObject> findObjects(GVRScene scene, float ox, float oy, float oz, float dx,

> +            float dy, float dz) {

> +        return Arrays.asList(pickObjects(scene, ox, oy, oz, dx, dy, dz));

> +    }

> +

> +    /**

> +     * Internal utility to help JNI add hit objects to the pick list.

> +     */

> +    public static GVRPickedObject makeHit(long colliderPointer, float distance, float hitx, float hity, float hitz)

Yup, it should work.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/Samsung/GearVRf/pull/621/files/6b44ced0b8880e5657bd8007fe45768d00d84573#r68316348>, or mute the thread<https://github.com/notifications/unsubscribe/AQw1Sx3kC2mVcL_u6HHZsjqTwJvpBqiDks5qOvdbgaJpZM4I6MsB>.